### PR TITLE
[dif] Update DIF milestones and requirements

### PR DESCRIPTION
**Pull request by timothytrippel**
The DIF milestones were out of date. This PR addresses  this (and a task in #10504). Specifically a few milestone descriptions / requirements were altered as discussed in #10504.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Review by msfschaffner — COMMENTED** [review]
I think this looks reasonable, but it would be good for SW folks to chime in on this one.

File: doc/project/sw_checklist.md.tpl [hunk 1]
```diff
@@ -17,59 +17,48 @@ Once done, please remove this comment before checking it in.
...
+<h3>S2</h3>
 
 Type           | Item                        | Resolution  | Note/Collaterals
 ---------------|-----------------------------|-------------|------------------
-Implementation | [DIF_FEATURES][]            | Not Started |
-Coordination   | [DIF_HW_USAGE_REVIEWED][]   | Not Started |
 Coordination   | [DIF_HW_FEATURE_COMPLETE][] | Not Started | [HW Dashboard]({{< relref "hw" >}})
-Implementation | [DIF_HW_PARAMS][]           | Not Started |
-Documentation  | [DIF_DOC_HW][]              | Not Started |
-Documentation  | [DIF_DOC_API][]             | Not Started |
```
Comment (mundaym) [inline doc/project/sw_checklist.md.tpl:45]: It looks like this checklist item was never documented but I do wonder if we need an explicit documentation check in S3. Do you think the code style check is enough?

**Review by mundaym — APPROVED** [review]

File: doc/project/sw_checklist.md.tpl (hunk 1 above)
Comment (timothytrippel) [reply doc/project/sw_checklist.md.tpl:45]: Yes, my thoughts exactly. Hence, I removed it. Sorry should have been more clear there.

File: doc/project/checklist.md [hunk 2]
```diff
@@ -619,36 +616,20 @@ This test will be shared with DV.
 
 Smoke tests are for diagnosing major issues in both software and hardware, and with this in mind, they should execute quickly.
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply doc/project/checklist.md:617]: Done.

**Pull request merged**